alacritty/src/window.rs

```diff
@@ -326,16 +338,18 @@ impl Window {
         self.window().request_user_attention(RequestUserAttentionType::Critical);
     }
 
-    #[cfg(windows)]
+    #[cfg(any(windows, not(feature = "x11")))]
     pub fn set_urgent(&self, _is_urgent: bool) {}
 
     pub fn set_outer_position(&self, pos: PhysicalPosition<i32>) {
         self.window().set_outer_position(pos);
     }
 
-    #[cfg(not(any(target_os = "macos", windows)))]
     pub fn x11_window_id(&self) -> Option<usize> {
-        self.window().xlib_window().map(|xlib_window| xlib_window as usize)
+        #[cfg(all(feature = "x11", not(any(target_os = "macos", windows))))]
+        return self.window().xlib_window().map(|xlib_window| xlib_window as usize);
+        #[cfg(any(target_os = "macos", windows, not(feature = "x11")))]
+        return None;
```

Comment:
Do we maybe want to make this and `wayland_display` into separate methods for `None`? I think that might be a bit cleaner. Either way it should be uniform and currently it's a bit of a mix between the two since `set_urgent` uses separate methods for example.